statsmodels/imputation/ros.py

```diff
@@ -0,0 +1,558 @@
+        result is left-censored. (i.e., True -> censored,
+        False -> uncensored)
+
+    Returns
+    -------
+    ranks : numpy.array
+        Array of ranks for the dataset.
+
+    """
+
+    ranks = (
+        df.assign(rank=1)
+          .groupby(by=[dl_idx, censorship])['rank']
+          .transform(lambda g: g.cumsum())
+    )
+    return ranks
+
+
+def _ros_plot_pos(row, censorship, cohn):
+    """
+    Compute the ROS plotting position for a result based on its rank,
+    censorship, detection limit index.
+
+    Parameters
+    ----------
+    row : pandas.Series or dict-like
+        Full observation (row) from a censored dataset. Requires a
+        'rank', 'detection_limit', and `censorship` column.
+    censorship : str
+        Name of the column in the dataframe that indicates that a
+        result is left-censored. (i.e., True -> censored,
+        False -> uncensored)
+    cohn : pandas.DataFrame
+        Dataframe of Cohn numbers.
+
+    Returns
+    -------
+    plotting_position : float
+
+    See also
+    --------
+    cohn_numbers
+
+    """
+
+    DL_index = row['det_limit_index']
+    rank = row['rank']
+    censored = row[censorship]
+
+    dl_1 = cohn.iloc[DL_index]
+    dl_2 = cohn.iloc[DL_index + 1]
+    if censored:
+        return (1 - dl_1['prob_exceedance']) * rank / (dl_1['ncen_equal']+1)
+    else:
+        return (1 - dl_1['prob_exceedance']) + (dl_1['prob_exceedance'] - dl_2['prob_exceedance']) * \
+                rank / (dl_1['nuncen_above']+1)
+
+
+def _norm_plot_pos(results):
+    """
+    Computes standard normal (Gaussian) plotting positions using scipy.
+
+    Parameters
+    ----------
+    results : array-like
+        Sequence of observed quantities.
+
+    Returns
+    -------
+    plotting_position : array of floats
+
+    """
+    ppos, sorted_res = stats.probplot(results, fit=False)
+    return stats.norm.cdf(ppos)
+
+
+def plotting_positions(df, censorship, cohn):
+    """
+    Compute the ROS plotting positions for results based on their rank,
+    censorship, detection limit index.
+
+    Parameters
+    ----------
+    df : pandas.DataFrame
+
+    censorship : str
+        Name of the column in the dataframe that indicates that a
+        result is left-censored. (i.e., True -> censored,
+        False -> uncensored)
+
+    cohn : pandas.DataFrame
+        Dataframe of Cohn numbers.
+
+    Returns
+    -------
+    plotting_position : array of float
+
+    See also
+    --------
+    cohn_numbers
+
+    """
+
+    plot_pos = df.apply(lambda r: _ros_plot_pos(r, censorship, cohn), axis=1)
+
+    # correctly sort the plotting positions of the ND data:
+    ND_plotpos = plot_pos[df[censorship]]
+    ND_plotpos.values.sort()
+    plot_pos.loc[df[censorship]] = ND_plotpos
+
+    return plot_pos
+
+
+def _ros_estimate(df, result, censorship, transform_in, transform_out):
+    """ Computed the estimated censored from the best-fit line of a
+    probability plot of the uncensored values.
```

Comment:
should be one line  (compute is redundant)
maybe "estimate censored observation from probability plot"

(just realized it's private, but still ...)